server/src/main/java/org/elasticsearch/discovery/zen/PublishClusterStateAction.java

```diff
@@ -397,6 +403,9 @@ protected void handleIncomingClusterStateRequest(BytesTransportRequest request,
         } catch (IncompatibleClusterStateVersionException e) {
             incompatibleClusterStateDiffReceivedCount.incrementAndGet();
             throw e;
+        } catch (Exception e) {
```

Comment:
This will also catch exceptions when calling `incomingClusterStateListener.onIncomingClusterState`, which in return calls `ZenDiscovery.validateIncomingState`. I think it's confusing to catch these as well here and erroneously reporting them here as deserialization errors.
Maybe it would be better to reduce the scope of the `catch` clause...

Reply:
That's a great comment. I missed that call. Let me see how I can shuffle things.

Reply:
Good catch @ywelsch.